### ISSUE TITLE
IDE0040 should not emit a warning for a partial declaration where a definition has a modifier

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#61826]
**Version Used**: VS 17.2

**Steps to Reproduce**:

1. Paste the following code:
```config
// In .editorconfig
dotnet_style_require_accessibility_modifiers = for_non_interface_members:warning
```
```csharp
public partial class Foo
{
}

partial class Foo
{
}
```
2. Roslyn will emit an IDE0040 warning on `partial class Foo`, saying there's no accessibility modifier.

**Expected Behavior**:

This case should not emit a warning. When a type has a partial declaration, it makes no sense to repeat all modifiers for all partial declarations, since they're shared anyway. This is not the same as a type _actually_ not having an accessibility modifier.

**Actual Behavior**:

An IDE0040 warning is emitted complaining about not having an accessibility modifier.

**Some thoughts**:

I reckon the warning should either be suppressed, or we should have a new `omit_if_default_or_present_in_partial_declaration` option for the .editorconfig setting. I really don't think Roslyn equating a missing accessibility modifier with a case where it's present in a partial declaration of the type is correct. The same applies to eg. not having to repeat other modifiers such as eg. `sealed`, `static`, etc. in a partial type declaration. I usually just put everything on the first one, as well as XML docs, and just use `inheritdoc` + `partial <TYPE>` in the other locations. This makes the code less verbose in other type declarations and avoids having to update multiple places if you need to change things.

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/52904.

**Comment by CyrusNajmabadi** [comment]
Closing as dupe.
